docs: improve language in Android section (#2890)
google/comprehensive-rust: src/android/interoperability/with-c.md

- The [`"C"` part][extern-abi] of the `extern` block tells Rust that `abs` can
  be called using the C [ABI] (application binary interface).

- The `safe fn abs` part tells Rust that `abs` is a safe function. By default,

>   extern functions are considered unsafe, but since `abs(x)` is valid for any

Reviewer: ```suggestion
  extern functions are unsafe, but since `abs(x)` can't trigger undefined behavior with any
```
